src/transformStateWithClones.js

```diff
@@ -7,7 +8,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    currentState = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties': {
+        const { extraData } = action;
+
+        Object.assign(currentState, extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        for (const key in currentState) {
+          delete currentState[key];
+        }
+        break;
+      }
+
+      default: {
+        return `Unknown action type: ${action.type}`;
```

Comment:
it`s right, it`s worked,  but in this task we need 'throw new Error'

Reply:
> it`s right, it`s work, but in this task we need 'throw new Error'

Good point